pyerrors/obs.py

```diff
@@ -279,14 +279,20 @@ def _compute_drho(i):
                 tmp = self.e_rho[e_name][i + 1:w_max] + np.concatenate([self.e_rho[e_name][i - 1::-1], self.e_rho[e_name][1:w_max - 2 * i]]) - 2 * self.e_rho[e_name][i] * self.e_rho[e_name][1:w_max - i]
                 self.e_drho[e_name][i] = np.sqrt(np.sum(tmp ** 2) / e_N)
 
-            _compute_drho(1)
+            # detect regular step size in irregular MC chain
+            gapsize = 1
```

Comment:
The combination of `np.min` and `np.diff` seems to be pretty fast even for large data sets. We could consider explicitly checking that gaps are consistent across replica

```python
gaps = []
for r_name in e_content[e_name]:
    if isinstance(self.idl[r_name], range):
        gaps.append(1)
    else:
        gaps.append(np.min(np.diff(self.idl[r_name])))

if not np.all(gaps == gaps[0]):
    raise Exception(f"Replica for ensemble {e_name} are not equally spaced.")
else:
    gapsize = gaps[0]
```
For two replica with `1000` samples each the overhead on my machine is 76.6 µs ± 2.17 µs compared to 34.7 µs ± 1.04 µs for your variant. The full `gamma_method` takes 11 ms ± 338 µs.

Reply:
That is indeed very fast, I'll include it.